### PR TITLE
feat: add support for updateDatabase in Cloud Spanner

### PR DESCRIPTION
**Pull request by aayushimalik**

### CODE REVIEW

File: google/cloud/spanner_v1/database.py [hunk 1]
```diff
@@ -29,6 +29,7 @@
 from google.api_core import gapic_v1
 from google.iam.v1 import iam_policy_pb2
 from google.iam.v1 import options_pb2
+from google.protobuf.field_mask_pb2 import FieldMask
 
 from google.cloud.spanner_admin_database_v1 import CreateDatabaseRequest
```
Comment (grace223) [inline google/cloud/spanner_v1/database.py:34]: Thanks

File: tests/unit/test_database.py [hunk 2]
```diff
@@ -17,8 +17,10 @@
 
 import mock
 from google.api_core import gapic_v1
+from google.cloud.spanner_admin_database_v1 import Database as DatabasePB
 from google.cloud.spanner_v1.param_types import INT64
```
Comment (grace223) [inline tests/unit/test_database.py:21]: Thanks

**Review by grace223 — APPROVED** [review]

File: google/cloud/spanner_v1/database.py [hunk 3]
```diff
@@ -488,6 +505,42 @@ def update_ddl(self, ddl_statements, operation_id=""):
         future = api.update_database_ddl(request=request, metadata=metadata)
         return future
 
+    def update(self):
+        """Update this database.
+
+        See
+        TODO: insert documentation once ready
```
Comment (rajatbhatta) [inline google/cloud/spanner_v1/database.py:512]:
```suggestion
        See  https://cloud.google.com/spanner/reference/rpc/google.spanner.admin.database.v1#google.spanner.admin.database.v1.DatabaseAdmin.UpdateDatabase
```

File: google/cloud/spanner_v1/database.py [hunk 4]
```diff
@@ -138,6 +142,7 @@ def __init__(
         encryption_config=None,
         database_dialect=DatabaseDialect.DATABASE_DIALECT_UNSPECIFIED,
         database_role=None,
+        drop_protection_enabled=False,
```
Comment (asthamohta) [inline google/cloud/spanner_v1/database.py:145]: This method is called from [instance.database()](https://github.com/googleapis/python-spanner/blob/main/google/cloud/spanner_v1/instance.py#L426), you need to add changes there as well

Comment (aayushimalik) [reply google/cloud/spanner_v1/database.py]: Done.

File: google/cloud/spanner_v1/database.py [hunk 5]
```diff
@@ -155,6 +160,8 @@ def __init__(
         self._encryption_config = encryption_config
         self._database_dialect = database_dialect
         self._database_role = database_role
+        self._enable_drop_protection = enable_drop_protection
+        self._reconciling = False
```
Comment (asthamohta) [inline google/cloud/spanner_v1/database.py:164]: I can't see any place where this value is updated?

File: google/cloud/spanner_v1/database.py [hunk 6]
```diff
@@ -488,6 +519,46 @@ def update_ddl(self, ddl_statements, operation_id=""):
...
+
+        .. note::
+
+            Updates the specified fields of a Cloud Spanner database. Currently,
+            only the `enable_drop_protection` field supports updates. To change
+            this value before updating, set it via
+
+            .. code:: python
+
+                database.enable_drop_protection = True
```
Comment (asthamohta) [inline google/cloud/spanner_v1/database.py:536]: Update documentation as per new implementation

File: tests/system/test_database_api.py [hunk 7]
```diff
@@ -562,3 +562,42 @@ def _unit_of_work(transaction, name):
...
+    shared_database.update(["enable_drop_protection"])
+
+def test_update_database_invalid(
+    not_emulator, shared_database
+):
+    shared_database.enable_drop_protection = True
+
+    # Empty `fields` is not supported.
+    with pytest.raises(exceptions.InvalidArgument):
+        shared_database.update([])
```
Comment (asthamohta) [inline tests/system/test_database_api.py:603]: nit: Add new line at end of file

Comment (aayushimalik) [reply tests/system/test_database_api.py]: Done.

File: google/cloud/spanner_v1/database.py (hunk 5 above)
Comment (aayushimalik) [reply google/cloud/spanner_v1/database.py:164]: This is updated at line 359 of this file, using the property setter.

File: google/cloud/spanner_v1/database.py (hunk 6 above)
Comment (aayushimalik) [reply google/cloud/spanner_v1/database.py:536]: I've made this a property and used the property setter for setting this (lines 357-360).

File: google/cloud/spanner_v1/database.py (hunk 5 above)
Comment (asthamohta) [reply google/cloud/spanner_v1/database.py:164]: I am referring to `reconciling`

**Review by asthamohta — APPROVED** [review]

**Comment by rajatbhatta** [comment]
Added samples from https://github.com/googleapis/python-spanner/pull/917 to here (for single merge). Both PRs are approved.

**Pull request merged**